app/js/components/orderEntry/addForm/AddForm.jsx

```diff
@@ -148,13 +156,39 @@ export class AddForm extends React.Component {
       fieldErrors: {
         ...this.state.fieldErrors, [name]: !item,
       },
+      options: {
+        ...this.state.options, [name]: options,
+      },
     }); }
   }
 
   handleChange = (event) => {
+    const { name, value } = event.target;
+    const {
+      drugDosingUnits, orderFrequencies, drugRoutes, durationUnits, drugDispensingUnits,
+    } = this.props.allConfigurations;
+    let options;
+    if (value && name === 'dosingUnit') {
```

Comment:
Noticing that the value is tested, it would be better practice to nest other conditions inside the condition that tests value. Thus, we would have something like this: 

```
if(value){
    switch (name) {}
}else{
}
```
With the above, you'd handle more conditions and would have a neater code.
@flacode

Reply:
@paradisekelechi ,  I wanted to use a switch statement, but I noticed that there is no default case therefore if else would be a better option.

Reply:
thank you for the resource